versions/development/SPEC.md

```diff
@@ -3126,6 +3030,20 @@ This function will return `false` if the argument is an unset optional value. It
   - ceil: Round **up** to the next higher integer
   - round: Round to the nearest integer based on standard rounding rules
 
+## String sep(String, Array[String])
+
+- Return the strings in the array concatenated, with the first parameter
+  between each.
+
+```
+sep(' ', ["a", "b", "c"])
+> "a b c"
+
+A = ["file_1", "file_2"]
+sep(' ', prefix('-i ', A))
+> "-i file_1 -i file_2"
+```
+
```

Comment:
I like the idea of converting this to a more delcarative engine function, as opposed to the old way. I wonder if we go through the trouble of making this an engine funciton, why not remane it to `join` as well. Join is much more indicative of the action taking place over `sep`

```
join('  ', ["a","b","c"])
>"a  b  c"
```

Reply:
join is familiar to python people, but it is also a famous stumbling block for newcomers - wdl's target audience. I am fine with it either way, but chose sep as status quo should be preferred.

Reply:
I don't really have a dog in this particular fight but my eyes perked up at the mention of Python. FWIW in the early days we tried to channel "what does Python do" when facing some situation where we couldn't decide what to do.